server/src/main/java/io/druid/metadata/SQLMetadataStorageActionHandler.java

```diff
@@ -206,306 +200,268 @@ public Boolean withHandle(Handle handle) throws Exception
-                       .execute() == 1;
+          @Override
+          public Boolean withHandle(Handle handle) throws Exception
+          {
+            return handle.createStatement(
+                String.format(
+                    "INSERT INTO %1$s (%2$s_id, lock_payload) VALUES (:entryId, :payload)",
+                    lockTable, entryTypeName
+                )
+            )
+                         .bind("entryId", entryId)
+                         .bind("payload", jsonMapper.writeValueAsBytes(lock))
+                         .execute() == 1;
+          }
         }
-      }
     );
   }
 
   public boolean removeLock(final long lockId)
   {
-    return retryingHandle(
-      new HandleCallback<Boolean>()
-      {
-        @Override
-        public Boolean withHandle(Handle handle) throws Exception
+    return connector.retryWithHandle(
+        new HandleCallback<Boolean>()
         {
-          return handle.createStatement(
-                      String.format(
-                          "DELETE FROM %s WHERE id = :id",
-                          lockTable
-                      )
-                  )
-                       .bind("id", lockId)
-                       .execute() == 1;
+          @Override
+          public Boolean withHandle(Handle handle) throws Exception
+          {
+            return handle.createStatement(
+                String.format(
+                    "DELETE FROM %s WHERE id = :id",
+                    lockTable
+                )
+            )
+                         .bind("id", lockId)
+                         .execute() == 1;
+          }
         }
-      }
     );
   }
 
   public boolean addLog(final String entryId, final LogType log)
   {
-    return retryingHandle(
-      new HandleCallback<Boolean>()
-      {
-        @Override
-        public Boolean withHandle(Handle handle) throws Exception
+    return connector.retryWithHandle(
+        new HandleCallback<Boolean>()
         {
-          return handle.createStatement(
-                      String.format(
-                          "INSERT INTO %1$s (%2$s_id, log_payload) VALUES (:entryId, :payload)",
-                          logTable, entryTypeName
-                      )
-                  )
-                       .bind("entryId", entryId)
-                       .bind("payload", jsonMapper.writeValueAsBytes(log))
-                       .execute() == 1;
+          @Override
+          public Boolean withHandle(Handle handle) throws Exception
+          {
+            return handle.createStatement(
+                String.format(
+                    "INSERT INTO %1$s (%2$s_id, log_payload) VALUES (:entryId, :payload)",
+                    logTable, entryTypeName
+                )
+            )
+                         .bind("entryId", entryId)
+                         .bind("payload", jsonMapper.writeValueAsBytes(log))
+                         .execute() == 1;
```

Comment:
This should actually never fail, since the log table just has an auto_increment primary key. If it retries when things actually did work the first time, we'll just get duplicate audit log entries. I think that's fine.